src/conf_macros.m4

```diff
@@ -558,21 +558,26 @@ AC_DEFUN([WITH_LIBNL],
 AC_DEFUN([WITH_CRYPTO],
     [ AC_ARG_WITH([crypto],
                   [AC_HELP_STRING([--with-crypto=CRYPTO_LIB],
-                                  [The cryptographic library to use (nss|libcrypto). The default is nss.]
+                                  [The cryptographic library to use (libcrypto|nss). The default is libcrypto (OpenSSL). NSS is deprecated.]
                                  )
                   ],
                   [],
-                  with_crypto=nss
+                  with_crypto=libcrypto
                  )
 
       cryptolib=""
       if test x"$with_crypto" != x; then
           if test x"$with_crypto" = xnss || \
           test x"$with_crypto" = xlibcrypto; then
               cryptolib="$with_crypto";
+              if test x"$with_crypto" = xnss; then
+                  AC_MSG_WARN([NSS is deprecated crypto backend and it support will be dropped in upcoming releases.])
+              fi
           else
               AC_MSG_ERROR([Illegal value -$with_crypto- for option --with-crypto])
           fi
+      else
+          cryptolib=libcrypto
```

Comment:
Not sure if we should set default value for $cryptolib. If may hide a bug in evaluating $with_crypto variable.

Reply:
This is to cover `--with-crypto=""` case. I think it is ok to treat this as default.

Reply:
Perhaps we could treat this as "Illegal value".
But just skipping check (as it was) doesn't make sense because without any backend build fails later.

Reply:
I think we should have two variants - option is empty or it has invalid value.
Still not sure if in those cases we should fail or use default value or just fail.

Reply:
> I think we should have two variants - option is empty or it has invalid value.

It is that way already.

> Still not sure if in those cases we should fail or use default value or just fail.

I would not bother way too much polishing this because at the (next+1) release we will drop NSS crypto backend (and thus this configure option) completely.

Reply:
Agree